contrib/isa_l/compression/source/compressor/config.cc

```diff
@@ -0,0 +1,75 @@
+#include "contrib/isa_l/compression/source/compressor/config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Compression {
+namespace Igzip {
+namespace Compressor {
+
+#ifndef IGZIP_DISABLED
+namespace {
+
+// Default memory level.
+const uint64_t DefaultMemoryLevel = 5;
+
+// Default compression window size.
+const uint64_t DefaultWindowBits = 15;
+
+// Default chunk size.
+const uint32_t DefaultChunkSize = 4096;
+} // namespace
+
+IgzipCompressorFactory::IgzipCompressorFactory(
+    const envoy::extensions::compression::compressor::igzip::v3alpha::Igzip& igzip)
+    : compression_level_(compressionLevelEnum(igzip.compression_level())),
+      memory_level_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(igzip, memory_level, DefaultMemoryLevel)),
+      window_bits_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(igzip, window_bits, DefaultWindowBits)),
+      chunk_size_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(igzip, chunk_size, DefaultChunkSize)) {}
+
+IgzipCompressorImpl::CompressionLevel IgzipCompressorFactory::compressionLevelEnum(
+    envoy::extensions::compression::compressor::igzip::v3alpha::Igzip::CompressionLevel
+        compression_level) {
+  switch (compression_level) {
+  case envoy::extensions::compression::compressor::igzip::v3alpha::Igzip::COMPRESSION_LEVEL_1:
+    return IgzipCompressorImpl::CompressionLevel::Level1;
+  case envoy::extensions::compression::compressor::igzip::v3alpha::Igzip::COMPRESSION_LEVEL_2:
+    return IgzipCompressorImpl::CompressionLevel::Level2;
+  case envoy::extensions::compression::compressor::igzip::v3alpha::Igzip::COMPRESSION_LEVEL_3:
+    return IgzipCompressorImpl::CompressionLevel::Level3;
+  default:
```

Comment:
I think we have miss some version here

Reply:
Yeah, I missed some items like speed/standard/best, but they will be removed next. Thanks!